snapshot/git/gitfiler/cloner.go

```diff
@@ -0,0 +1,45 @@
+package gitfiler
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+
+	"github.com/scootdev/scoot/os/temp"
+	"github.com/scootdev/scoot/snapshot/git/repo"
+)
+
+// A Reference Repository is a way to clone repos locally so that the clone takes less time and disk space.
+// By passing --reference <local path> to a git clone, the clone will not copy the whole ODB but instead
+// hardlink. This means the clone is much faster and also takes very little extra hard disk space.
+// Cf. https://git-scm.com/docs/git-clone
+
+// cloner clones a repo using --reference based on a reference repo
+type refCloner struct {
+	refPool   *RepoPool
```

Comment:
I think we need to revisit the names at some point, these are getting pretty long. I may be confused and I think requiring an [expandable] pool with a static single repo contributes to that.